functional-tests/src/test/groovy/spec/Tailings.groovy

```diff
@@ -38,5 +50,46 @@ class Tailings extends GebReportingSpec {
         then: "the reports are visible"
         assert tailingsTab.document0Name != null
     }
+
+    //ToDo: THIS TEST ONLY WORKS IN CHROME AND CHROME HEADLESS.  WORK IN GebConfig to fix other browsers
+    def "Scenario: User is able to upload a TSF Report"(){
+        when: "User navigates to the TSF tab and clicks the upload icon"
+        tailingsTab.tabSelect.click()
+
+        and: "User opens modal and uploads a valid file type"
+        def uploadedFile = dir.newFile(Const.TEST_FILE_NAME) << Const.TEST_FILE_CONTENT
+        tailingsTab.addTailingsDocButtons[0].click()
+        tailingsTab.uploadField = uploadedFile.absolutePath
+
+        then: "The doc upload complete message is shown"
+        assert tailingsTab.uploadCompleteMessage != null
+
+    }
+
+    //ToDo: THIS TEST ONLY WORKS IN CHROME AND CHROME HEADLESS.  WORK IN GebConfig to fix other browsers
+    def "Scenario: User is able to download a TSF Report"() {
+        when: "User navigates to the TSF tab and clicks the download icon"
+        tailingsTab.tabSelect.click()
+
+        and: "User opens a file in the folder specified in GebConfig"
+        tailingsTab.downloadLink[0].click()
+
+        def file = new File(Const.DOWNLOAD_PATH+'/'+Const.TEST_FILE_NAME)
+        //allow time for the file to be created in the DOWNLOAD_PATH
+        //throw an error if it takes more than 20sec
```

Comment:
Do we have any idea of how long this **can** take and then still be successful? Our other timeouts in the functional tests are 30 seconds, if I remember correctly.

Reply:
I gave it a 20 second loop somewhat arbitrarily.  If it takes longer, to download than that, something has gone wrong.  I wanted it to be less than 30 seconds so there would be clear error message about what went wrong.